src/Campaigns/Blocks/CampaignGoal/utils.ts

```diff
@@ -0,0 +1,35 @@
+import {__} from '@wordpress/i18n';
+
+export const getGoalDescription = (goalType: string) => {
+    switch (goalType) {
+        case 'amount':
+            return __('Amount raised', 'give');
+        case 'donations':
+            return __('Number of donations', 'give');
+        case 'donors':
+            return __('Number of donors', 'give');
+        case 'amountFromSubscriptions':
+            return __('Recurring amount raised', 'give');
+        case 'subscriptions':
+            return __('Number of recurring donations', 'give');
+        case 'donorsFromSubscriptions':
+            return __('Number of recurring donors', 'give');
+    }
+}
+
+
+export const getGoalFormattedValue = (goalType: string, value: number) => {
+    switch (goalType) {
+        case 'amount':
+        case 'amountFromSubscriptions':
+            const currency = new Intl.NumberFormat(navigator.language, {
+                style: 'currency',
+                currency: window.GiveCampaignOptions.currency,
```

Comment:
Feel free to use the new helpers 
- [getCampaignDetailsWindowData](https://github.com/impress-org/givewp/blob/de80d9b6ca7927e6acd018d7208c2ab1afe6b54c/src/Campaigns/resources/admin/common/getCampaignDetailsWindowData.ts)
- [amountFormatter](https://github.com/impress-org/givewp/blob/de80d9b6ca7927e6acd018d7208c2ab1afe6b54c/src/Campaigns/resources/utils.ts#L29)

```js
const {currency} = getCampaignDetailsWindowData();
const currencyFormatter = amountFormatter(currency);

currencyFormatter.format(value);
```

Reply:
This looks handy, but it works only on the Campaign Details page. We should be able to use it everywhere we can add campaign blocks. I'll refactor it.

Reply:
I did a complete refactor https://github.com/impress-org/givewp/pull/7702/commits/da65f7d87874209e69bc42e76f7bba1fcd7a6a22

Reply:
Haha works for me!